catalogue/webapp/services/prismic/index.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  Story,
+  PrismicApiError,
+  StoryResultsList,
+  PrismicResponseStory,
+} from '@weco/common/model/story';
+import fetch, { Response } from 'node-fetch';
+import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';
+import { HttpsAgent as Agent } from 'agentkeepalive';
+
+export type PrismicQueryProps = {
+  query?: string | string[]
+  pageSize?: number;
+};
+
+export const prismicApiError = (): PrismicApiError => ({
+  errorType: 'http',
+  httpStatus: 500,
+  label: 'Internal Server Error',
+  description: '',
+  type: 'Error',
+});
+
+const agentKeepAlive = new Agent({
+  keepAlive: true,
+  freeSocketTimeout: 1000 * 59, // 1s less than the akka-http idle timeout
+});
+
+// TODO: move Prismic fetch function out to common and use it here and in server-data/prismic
+// To authenticate a request to the Prismic API graphql endpoint you need a Prismic-ref value that refreshes every 30 seconds.
+// The ref can be used in a 'stale' state for 5 minutes before it expires
+// https://community.prismic.io/t/for-how-long-is-it-safe-to-cache-a-prismic-api-ref/5962
+export const prismicRefFetch = (
+  url: string,
+  options?: Record<string, string>
+): Promise<Response> => {
+  return fetch(url, { ...options, agent: agentKeepAlive });
+};
+
+// Prismic API graphql endpoint expects query requests via GET method
+// This means we need to be able to pass a body in the request, something that is not possible with fetch
+// As a workaround I have used axios to make the request
+export const prismicFetch = (
+  options?: AxiosRequestConfig
+): Promise<AxiosResponse> => {
+  return axios({ ...options });
+};
+
+export async function prismicQuery(
+  endpoint: string,
+  { query, pageSize }: PrismicQueryProps
+): Promise<StoryResultsList<Story> | PrismicApiError> {
+  const graphQuery = `query {
+  allArticless(fulltext: "${query}" sortBy: title_ASC first: ${pageSize}) {
+  edges {
+      node {
+        title
+        _meta { id }
+        contributors {
+          contributor {
+            ...on People {
+              name
+            }
+          }
+        }
+        body {
+          ...on ArticlesBodyStandfirst {
+            primary {
+              text
+            }
+          }
+        }
+        promo {
+          ...on ArticlesPromoEditorialimage {
+            primary {
+              image
+              link
+              caption
+            }
+          }
+        }
+      }
+    }
+  }
+}`;
+  const url = `https://wellcomecollection.prismic.io/graphql?query=${graphQuery}`;
+  const fetchRefUrl = 'https://wellcomecollection.prismic.io/api/v2';
```

Comment:
These two const are only used once (I think?), I wonder if it'd be more readable to move them where they are used?